pkg/controller/backup_sync_controller.go

```diff
@@ -103,35 +132,36 @@ func (c *backupSyncController) run() {
 	locations = orderedBackupLocations(locations, c.defaultBackupLocation)
 
 	pluginManager := c.newPluginManager(c.logger)
+	defer pluginManager.CleanupClients()
 
 	for _, location := range locations {
 		log := c.logger.WithField("backupLocation", location.Name)
-		log.Info("Syncing backups from backup location")
 
 		backupStore, err := c.newBackupStore(location, pluginManager, log)
 		if err != nil {
 			log.WithError(err).Error("Error getting backup store for location")
 			continue
 		}
 
-		backupsInBackupStore, err := backupStore.ListBackups()
+		syncTime := time.Now().UTC()
+		ok, revision := shouldSync(location, syncTime, backupStore, log)
+		if !ok {
+			continue
+		}
+
+		res, err := backupStore.ListBackups()
 		if err != nil {
 			log.WithError(err).Error("Error listing backups in backup store")
 			continue
 		}
+		backupStoreBackups := sets.NewString(res...)
+		log.WithField("backupCount", len(backupStoreBackups)).Info("Got backups from backup store")
 
-		log.WithField("backupCount", len(backupsInBackupStore)).Info("Got backups from backup store")
-
-		cloudBackupNames := sets.NewString()
-		for _, cloudBackup := range backupsInBackupStore {
-			log = log.WithField("backup", kube.NamespaceAndName(cloudBackup))
-			log.Debug("Checking cloud backup to see if it needs to be synced into the cluster")
+		for backupName := range backupStoreBackups {
+			log = log.WithField("backup", backupName)
+			log.Debug("Checking backup store backup to see if it needs to be synced into the cluster")
```

Comment:
Is this debug msg correct?

Reply:
I think so? which part were you questioning